libs/commons/Logs_helpers.ml

```diff
@@ -1,34 +1,35 @@
-(* ANSI escape sequences for colored output, depending on log level *)
-let color level =
-  match level with
-  | Logs.Warning -> Some "33" (*yellow*)
-  | Logs.Error -> Some "31" (*red*)
-  | _else -> None
+(** This global is used by our reporter to do the diff between the last
+    log we outputed and the current one. Actually, the implementation is a
+    bit dumb and probably show weird metrics when we use [lwt]. For such
+    case, it's better to add a _counter_ and use the tag mechanism to really
+    show right metrics. *)
 
-(* print an ANSI escape sequence - not worth to use an extra library
-   (such as ANSIterminal) for this *)
-let pp_sgr ppf style =
-  Format.pp_print_as ppf 0 "\027[";
-  Format.pp_print_as ppf 0 style;
-  Format.pp_print_as ppf 0 "m"
+let value = ref 0.
+
+let now () =
+  Mtime.to_uint64_ns (Mtime_clock.now ())
+  |> Duration.to_sec_64 |> Int64.to_float
 
 (* log reporter *)
-let reporter ?(dst = Format.err_formatter) () =
+let reporter ?(with_timestamp = false) ?(dst = Fmt.stderr) () =
   let report _src level ~over k msgf =
-    let pp_style, style, style_off =
-      match color level with
-      | None -> ((fun _ppf _style -> ()), "", "")
-      | Some x -> (pp_sgr, x, "0")
-    in
     let k _ =
       over ();
       k ()
     in
     let r =
-      msgf @@ fun ?header:_ ?tags:_ fmt ->
-      Format.kfprintf k dst ("@[%a" ^^ fmt ^^ "@]@.") pp_style style
+      msgf @@ fun ?header ?tags:_ fmt ->
+      match with_timestamp with
+      | true ->
+          let current = now () in
+          Fmt.kpf k dst
+            ("@[[+%010.2fms]%a: " ^^ fmt ^^ "@]@.")
+            (current -. !value) Logs_fmt.pp_header (level, header)
+      | _ ->
+          Fmt.kpf k dst
+            ("@[%a" ^^ fmt ^^ "@]@.")
+            Logs_fmt.pp_header (level, header)
```

Comment:
when we're not in --debug, we need to keep the exact same log format as before, otherwise
we get failures in 'make osempass' like the one you get right now.
you need to replace Logs_fmt.pp_header with the previous pp_style that was there before